src/hash.rs

```diff
@@ -132,6 +132,16 @@ where
         self.hash()
     }
 
+    /// Returns hash value for MT leaf (prefix 0x00).
+    #[inline]
+    fn multi_leaf(&mut self, leaves: Vec<T>) -> T {
```

Comment:
this might be a bit tricky, but could help quite well with perf. You should be able to make this generic over this size by using https://docs.rs/generic-array/ instead of a vec